chemper/smirksify.py

```diff
@@ -166,27 +185,137 @@ def types_match_reference(self, current_types=None):
         type_matches, checks_pass = match_reference(current_assignments, self.cluster_dict)
         return type_matches, checks_pass
 
-    def print_smirks(self, smirks_list=None):
+    def remove_one_sub_dec(self, input_ors, ref_idx):
+        """
+        Remove one ORdecorator from the specified index
+        # i.e. [(#6, [X4, +0]), (#7, [X3]) ] --> [(#6, [+0]), (#7, [X3]) ]
+        """
+        ors = copy.deepcopy(input_ors)
+        # remove one or decorator from a single type
+        ref_or = ors[ref_idx]
+        decs = ref_or[1]
+        decs.remove(random.choice(decs))
+        ors[ref_idx] = (ref_or[0], decs)
+        return ors
+
+    def remove_ref_sub_decs(self, input_ors, ref_idx):
+        """
+        Remove all of the ORdecorators at the specified index
+        i.e. [(#6, [X4, +0]), (#7, [X3]) ] --> [(#6, []), (#7, [X3]) ]
+        """
+        ors = copy.deepcopy(input_ors)
+        # remove all decorators on one OR type
+        ref_or = ors[ref_idx]
+        ors[ref_idx] = (ref_or[0], list())
+        return ors
+
+    def remove_ref(self, input_decs, ref_idx):
         """
-        Prints out the current or provided smirks list
-        in a table like format
+        Remove the decorator at the referenced index
```

Comment:
Similar comment to before, it would be helpful if you included definition of the inputs in the function and also an example.